### PR TITLE
[ci] Roll minimum allowable Flutter to 3.10

### PR DESCRIPTION
**Pull request by stuartmorgan**
Updates the repo for Flutter 3.16, making the minimum version that packages are allowed to support 3.10, following the process in https://github.com/flutter/flutter/wiki/Updating-Packages-repo-for-a-stable-release

Per https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version this does not bump versions.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
LGTM

**Review by tarrinneal — APPROVED** [review]
tldr

**Review by ditman — APPROVED** [review]
LGTM!

(Also apologies for rolling `stable` but not doing the rest of the stuff 😱)

File: packages/file_selector/file_selector_web/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Updates minimum supported SDK version to Flutter 3.10/Dart 3.0.
+
```
Comment (ditman) [inline packages/file_selector/file_selector_web/CHANGELOG.md:4]: This one has already been migrated [here](https://github.com/flutter/packages/commit/42dbb7573aa2f8263483f905114dbfaab1810b46#diff-6127965c509bfd98a9a6f317d0dd256ce511d6257aa4912ddee1726230ec0392) (to move from dart:html to package:web), I think this changelog may be reverted.
